dagshub/streaming/filesystem.py

```diff
@@ -326,12 +326,15 @@ def cleanup(self):
             DagsHubFilesystem.already_mounted_filesystems.pop(self.project_root)
 
     def _parse_path(self, file: Union[str, PathLike, int]) -> DagshubPath:
-        orig_path = Path(file)
         if isinstance(file, int):
-            return DagshubPath(self, None, None, orig_path)
+            return DagshubPath(self, None, None, Path(file))
         if file == "":
-            return DagshubPath(self, None, None, orig_path)
-        abspath = Path(os.path.abspath(file))
+            return DagshubPath(self, None, None, Path(file))
+
+        expanded = os.path.expanduser(file)
+        orig_path = Path(expanded)
```

Comment:
1. You intentionally don't expand the `orig_path` before passing it into the `DagshubPath`? Just so I understand why. Maybe a comment explaining it?
2. I think you introduced a bug here. The default `open` doesn't `expanduser` on the supplied path, from what I can gather. But now, with this change, every call to `open` will pass through us and get parsed here and then you use the returned absolute path in the call to `__open`. So I think instead, to be really safe, you should make sure to always call the underlying patched functions with exactly the same args as you were called with, making sure they're not modified in any way

Reply:
I guess comment 2 is also relevant for `__chdir` and `__listdir` etc.

Reply:
[duplicating in-person chat]
> You intentionally don't expand the orig_path before passing it into the DagshubPath? Just so I understand why. Maybe a comment explaining it?
Yes, this is because this original path is later reused in functions that have different behavior depending on the accessed paths (notable example: `listdir`, that returns different paths depending on whether passed an absolute or relative path)

> 2.
Yea, you're right, I actually shouldn't be expanding it at all probably.